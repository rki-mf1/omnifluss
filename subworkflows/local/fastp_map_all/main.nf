include { BWA_INDEX              } from '../../../modules/nf-core/bwa/index/main'
include { FASTQ_ALIGN_BWA        } from '../../nf-core/fastq_align_bwa/main'

workflow FASTP_MAP_ALL {
    take:
    tools           // string
    ch_reads        // channel: [ val(meta), fastq ]
    reference       // channel: [ val(meta), fasta ]

    main:
    ch_index = Channel.empty()
    ch_bam_orig = Channel.empty()
    ch_bam = Channel.empty()
    ch_csi = Channel.empty()
    ch_stats = Channel.empty()
    ch_flagstat = Channel.empty()
    ch_idxstats = Channel.empty()
    ch_versions = Channel.empty()

    if (tools.split(',').contains('bwa')) {

        BWA_INDEX(
            reference   // channel: [ val(meta), fasta ]
        )
        ch_index = BWA_INDEX.out.index
        ch_versions = ch_versions.mix(BWA_INDEX.out.versions)

        FASTQ_ALIGN_BWA(
            ch_reads,   // channel: [ val(meta), fastq ]
            ch_index,   // channel: [ val(meta), index ]
            true,
            reference   // channel: [ val(meta), fasta ]
        )
        ch_bam_orig = FASTQ_ALIGN_BWA.out.bam_orig
        ch_bam = FASTQ_ALIGN_BWA.out.bam
        ch_csi = FASTQ_ALIGN_BWA.out.csi
        ch_stats = FASTQ_ALIGN_BWA.out.stats
        ch_flagstat = FASTQ_ALIGN_BWA.out.flagstat
        ch_idxstats = FASTQ_ALIGN_BWA.out.idxstats
        ch_versions = ch_versions.mix(FASTQ_ALIGN_BWA.out.versions.first())

    }
    emit:
    ch_mapping        = ch_bam
    versions          = ch_versions

}

