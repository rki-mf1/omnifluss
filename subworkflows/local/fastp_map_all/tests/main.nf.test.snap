{
    "sarscov2 - single_end - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.sorted.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f59ef8409a8c5616424cbcd4ce1d9d3e",
                    "versions.yml:md5,fcb9654468b0268146c6711dea5f6758"
                ],
                "ch_mapping": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.sorted.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f59ef8409a8c5616424cbcd4ce1d9d3e",
                    "versions.yml:md5,fcb9654468b0268146c6711dea5f6758"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-07-11T10:38:45.507718474"
    },
    "sarscov2 - paired_end - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sorted.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f59ef8409a8c5616424cbcd4ce1d9d3e",
                    "versions.yml:md5,fcb9654468b0268146c6711dea5f6758"
                ],
                "ch_mapping": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sorted.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f59ef8409a8c5616424cbcd4ce1d9d3e",
                    "versions.yml:md5,fcb9654468b0268146c6711dea5f6758"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-07-11T10:39:08.221218517"
    }
}